src/core/api/coap_api.cpp

```diff
@@ -144,6 +144,11 @@ otCoapCode otCoapMessageGetCode(const otMessage *aMessage)
     return static_cast<otCoapCode>(AsCoapMessage(aMessage).GetCode());
 }
 
+void otCoapMessageSetCode(otMessage *aMessage, otCoapCode aCode)
+{
+    AsCoapMessage(aMessage).SetCode(static_cast<Coap::Code>(aCode));
```

Comment:
```suggestion
    AsCoapMessage(aMessage).SetCode(MapEnum(aCode));
```

Suggest using `MapEnum(aCode)` instead of `static_cast<Coap::Code>(aCode)`.

`MapEnum` is itself defined in `as_core_type.hpp` and it converts the related `enum` types to each other. 
- It is simpler and safer to use in the sense that it only works between `enum` types that we explicitly relate using `DefineMapEnum()` (e.g., in this csae`otCoapCode` and `ot::Coap::Code` which are related in `coap_message.hpp`). 
- If we try to `MapEnum` between unrelated types, we get a compile error, while `static_cast` allows any `enum` type to be cast to any other type.

Reply:
Thanks for pointing out. That indeed is a lot nicer. Adopted